traveler-ios-ui/TravelerKitUI/ViewControllers/WishlistResultViewController.swift

```diff
@@ -0,0 +1,241 @@
+        default:
+            Log("Unknown segue", data: segue, level: .warning)
+            break
+        }
+    }
+
+    // MARK: UICollectionViewDataSource
+
+    override open func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+
+    override open func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return wishlistResult?.total ?? 0
+    }
+
+    override open func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let catalogItem = wishlistResult!.items[indexPath.row]
+
+        switch catalogItem {
+        case .none:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: loadingCollectionCellIdentifier, for: indexPath)
+            cell.contentView.subviews.forEach({ $0.startShimmering() })
+            return cell
+        case .some(let item):
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: wishlistCellIdentifier, for: indexPath) as! WishlistCell
+
+            cell.titleLabel.text = item.title
+            cell.descriptionLabel.text = item.subTitle
+            cell.priceLabel.text = "Price starting at: " + (item.price.localizedDescriptionInBaseCurrency ?? "")
+            cell.isAvailable = item.isAvailable
+
+            if let url = item.imageURL {
+                AssetManager.shared.loadImage(with: url) { [weak cell] (image) in
+                    cell?.thumbnailImageView.image = image
+                }
+            } else {
+                cell.thumbnailImageView.image = nil
+            }
+
+            cell.delegate = self
+
+            return cell
+        }
+    }
+
+    // MARK: UICollectionViewDelegate
+    override open func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        let catalogItem = wishlistResult!.items[indexPath.row]
+
+        return catalogItem != nil ? true : false
+    }
+
+    override open func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard let item = wishlistResult!.items[indexPath.row] else {
+            Log("Unknown item selected", data: nil, level: .error)
+            return
+        }
+
+        if item.isAvailable == true {
+            selectedCatalogItem = item
+            collectionView.deselectItem(at: indexPath, animated: true)
+            performSegue(withIdentifier: "itemSegue", sender: nil)
+        } else {
+            let alert = UIAlertController(title: "This item is no longer available. Do you want to remove it from your wishlist?", message: nil, preferredStyle: .alert)
+            let noAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+            let yesAction = UIAlertAction(title: "Yes", style: .default) { [unowned self] _ in
+                self.collectionView.performBatchUpdates({
```

Comment:
Can you make sure Android take into account this edge case?

+1

Reply:
acknowledged